streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -261,7 +293,7 @@ public String toString() {
      * @throws IllegalStateException if instance is currently running
      */
     public void cleanUp() {
-        if (state == RUNNING) {
+        if (state == KafkaStreamsState.RUNNING || state == KafkaStreamsState.REBALANCING) {
```

Comment:
The original check meant that streams has been running (not just created). In our context, this means just running or rebalancing.